xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -350,36 +350,91 @@ static final class StaticStrideScheduler {
     private final short[] scaledWeights;
     private final AtomicInteger sequence;
     private static final int K_MAX_WEIGHT = 0xFFFF;
+    // Assuming the mean of all known weights is M, StaticStrideScheduler will cap
+    // from above all known weights that are bigger than M*kMaxRatio (to
+    // M*kMaxRatio).
+    //
+    // This is done to limit the number of rounds for picks.
+    private static final double K_MAX_RATIO = 10;
+
+    // Assuming the mean of all known weights is M, StaticStrideScheduler will cap
+    // from below all known weights to M*kMinRatio.
+    //
+    // This is done as a performance optimization for edge cases when channels with
+    // large weights are non-accepting (and thus WeightedRoundRobin will retry
```

Comment:
This might not impact gRPC. The list passed into this scheduler has already filtered out the non-accepting subchannels (and the definition of non-accepting is not quite the same). That means the worst-case time is O(n), because at the very least we'd pick the subchannel with the max weight.

It is possible in gRPC for a transport not to accept an RPC, but in that case the RPC is queued and we wait for the LB to notice the subchannel is no longer available. That will then generate a new scheduler with a new max weight.

Reply:
Aha, so I guess we wouldn't run into the non-accepting, infinite loop issue that Stubby did, but this is still useful to optimize for cases where there's a great difference in weights?

Reply:
I think it is useful to put in place a bound.